test/unit/lib/static/modules/actions.js

```diff
@@ -131,4 +131,50 @@ describe('lib/static/modules/actions', () => {
             );
         });
     });
+
+    describe('runFailedTests', () => {
+        const mkBrowserGivenResult = (status) => (name) => {
+            return mkBrowserResult({
+                name,
+                status,
+                result: mkTestResult({
+                    name,
+                    status
+                })
+            });
+        };
+
+        it('should run all failed tests for suite with children and browsers', async () => {
+            const failed = [
+                mkSuite({
+                    suitePath: ['suite1'],
+                    children: [
+                        mkSuite({
+                            suitePath: ['suite1', 'suite2'],
+                            children: [
+                                mkState({
+                                    suitePath: ['suite1', 'suite2', 'state'],
+                                    browsers: [mkBrowserGivenResult(FAIL)('yabro'), mkBrowserGivenResult(ERROR)('chrome')]
+                                })
+                            ]
+                        })
+                    ]
+                })
+            ];
+
+            await runFailedTests(failed)(dispatch);
+
+            assert.calledWith(
+                axios.post,
+                sinon.match.any,
+                sinon.match(formattedFails => {
+                    assert.lengthOf(formattedFails, 1);
+                    assert.lengthOf(formattedFails[0].browsers, 2);
+                    assert.equal(formattedFails[0].browsers[0].name, 'yabro');
+                    assert.equal(formattedFails[0].browsers[1].name, 'chrome');
+                    return true;
```

Comment:
а эт че?

Reply:
`Custom matchers are created with the sinon.match factory which takes a test function and an optional message.

The test function takes a value as the only argument, returns true if the value matches the expectation and false otherwise. `

Reply:
Зачем столько проверок здесь? Нужно было просто проверить, что запустился тест в двух браузерах и все.

Reply:
Otherwise you get a message that tells you nothing. Something like 'TypeError: Cannot read property 'browsers' of undefined` or `TypeError: Cannot read property 'name' of undefined`